src/bregman.jl

```diff
@@ -111,19 +111,20 @@ function bregman(funobj::Function, x::AbstractArray{T}, options::BregmanParams,
         d .= -TD*g
         # Step length
         t = (options.spg && i> 1) ? T(dot(x-xold, x-xold)/dot(x-xold, g-gold)) : T(options.alpha*f/norm(d)^2)
+        t = abs(t)
+        mul!(d, d, t)
 
         # Anti-chatter
         if options.antichatter
-            @. tk = tk - sign.(d)
+            @. tk = tk - sign(d)
             # Chatter correction
             inds_z = findall(abs.(z) .> λ)
-            mul!(d, d, t)
-            d[:] .*= abs.(tk[:])/i
+            @views d[inds_z] .*= abs.(tk[:])/i
         end
         # Update z variable
         @. z = z + d
         # Get λ at first iteration
-        i == 1 && (λ = T(quantile(abs.(z), options.quantile)))
+        i == 1 && (λ = abs(T(quantile(abs.(z), options.quantile))))
```

Comment:
don't see why would need `abs` here, it's `quantile(abs(z))` it's positive.

Reply:
When T is complex then it's complex